crates/chain/error.rs

```diff
@@ -0,0 +1,34 @@
+use thiserror::Error;
+
+use ethereum_rust_evm::EvmError;
+use ethereum_rust_storage::error::StoreError;
+
+#[derive(Debug, Error)]
+pub enum ChainError {
+    #[error("Invalid Block: {0}")]
+    InvalidBlock(InvalidBlockError),
+    #[error("Parent block not found")]
+    ParentNotFound,
+    //TODO: If a block with block_number greater than latest plus one is received
+    //maybe we are missing data and should wait for syncing
+    #[error("Block number is greater than the latest plus one")]
+    NonCanonicalBlock,
+    #[error("DB error: {0}")]
+    StoreError(StoreError),
+    #[error("EVM error: {0}")]
+    EvmError(EvmError),
```

Comment:
```suggestion
    #[error("DB error: {0}")]
    StoreError([#from] StoreError),
    #[error("EVM error: {0}")]
    EvmError([#from] EvmError),
```
With this attribute we can avoid explicitly mapping between StoreError/EvmError and ChainError

Reply:
That's great, thanks!